cloudinit/config/schema.py

```diff
@@ -109,6 +155,164 @@ def is_schema_byte_string(checker, instance):
     ) or isinstance(instance, (bytes,))
 
 
+def _add_deprecation_msg(
+    description: Optional[str] = None,
+    deprecated_time: Optional[str] = None,
+):
+    msg = "DEPRECATED."
+    if deprecated_time:
+        if msg:
+            msg += " "
+        msg += f"Dropped in {deprecated_time}."
+    if description:
+        if msg:
+            msg += " "
+        msg += f"{description}"
+    return msg
+
+
+def _validator_deprecated(
+    _validator,
+    deprecated: bool,
+    _instance,
+    schema: dict,
+    error_type: Type[Exception],
+):
+    """Jsonschema validator for `deprecated` items.
+
+    It raises a instance of `error_type` if deprecated that must be handled,
+    otherwise the instance is consider faulty.
+    """
+    if deprecated:
+        deprecated_time = schema.get(DEPRECATED_TIME_KEY)
+        description = schema.get("description")
+        msg = _add_deprecation_msg(description, deprecated_time)
+        yield error_type(msg)
+
+
+def _anyOf(validator, anyOf, instance, _schema, error_type: Type[Exception]):
```

Comment:
Is this, along with _oneOf and _allOf, using the same implementation in the base but just adding the yield?

Reply:
1. `_validator_deprecated` yields a `SchemaDeprecationError` if `deprecated: true` is found in a sub-schema.
2. `_anyOf` same as base plus:
   * If the instance matches a subschema then yields `SchemaDeprecationError`s if present.
3. `_oneOf`: similar as `_anyOf`
4. `_allOf`: base + yield all `SchemaDeprecationError`s present in sub-schemas.
5. `is_valid`: base + ignore `SchemaDeprecationError`s.

I did try to document these functions, please let me know if you think we need to improve it.